Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+   * @var int
+   */
+  private $limit;
+
+  /**
+   * OFFSET.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var int
+   */
+  private $offset;
+
+  /**
+   * Query result.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var mixed
+   */
+  private $result;
+
+  /**
+   * Set target table.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $name Table name.
+   * @return self
+   */
+  public function table($name) {
+    // Verify that $name is a string.
+    if (!is_string($name))
+      throw InvArgExcp::typeMismatch('table name', $name, 'String or array');
+
+    $this->table = $name;
+    return $this;
+  }
+
+  /**
+   * Set SELECT fields.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array|string $field Field name.
+   * @return self
+   */
+  public function select($fields) {
+    $fields = is_array($fields) ? $fields : func_get_args();
+    $this->select = array_filter($fields, 'is_string');
+    $this->type = 'select';
+
+    return $this;
+  }
+
+  /**
+   * Set UPDATE fields and values.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array $values Associative array `key => value`, where `key` is the
+   * name of the field and `value` is the new value.
+   * @return self
+   *
+   * @throws SqueezyWeb\Exceptions\InvalidArgumentException if field names in
+   * $values aren't strings.
+   * @throws \RuntimeException if values in $values aren't scalars.
+   */
+  public function update(array $values) {
+    foreach ($values as $field => &$value) {
+      if (!is_string($field))
+        throw InvArgExcp::typeMismatch('field name', $field, 'String');
+      try {
+        $value = self::correctValue($value, 'update');
+      } catch (Exception $e) {
+        throw $e;
+      }
+    }
+
+    $this->update = $values;
+    $this->type = 'update';
+
+    return $this;
+  }
+
+  /**
+   * Set COUNT.
+   *
+   * Set COUNT to all field or to some specific fields (passing an array
+   * containing strings).
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array $count Optional. List of fields to which apply the COUNT
+   * selector. Default: empty array.
+   * @return self
+   */
+  public function count(array $fields = array()) {
+    if (empty($fields))
+      $this->count = true;
+    else
+      $this->count = array_filter($fields, 'is_string');
+    $this->type = 'select';
+
+    return $this;
+  }
+
+  /**
+   * Set INSERT fields and values.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array $values Associative array `key => value`, where `key` is the
+   * field and `value` is the new value.
+   * @return self
+   *
+   * @throws SqueezyWeb\Exceptions\InvalidArgumentException if field names in
+   * $values aren't strings.
+   * @throws \RuntimeException if values in $values aren't scalars.
+   */
+  public function insert(array $values) {
+    foreach ($values as $field => &$value) {
+      if (!is_string($field))
```

Comment:
Same as for [MySQLQuery::update()](https://github.com/SqueezyWeb/Database/pull/3/files#r51755662)